### PR TITLE
Update the ref with 2.4 branch for OSD.

### PR DESCRIPTION
**Pull request by zelinh**
Signed-off-by: Zelin Hao <zelinhao@amazon.com>

### Description
Change the ref for OSD from commit ID to `2.4` branch.

### Issues Resolved
Part of https://github.com/opensearch-project/opensearch-build/issues/2649

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: manifests/2.4.0/opensearch-dashboards-2.4.0.yml [hunk 1]
```diff
@@ -9,49 +9,49 @@ ci:
 components:
   - name: OpenSearch-Dashboards
     repository: https://github.com/opensearch-project/OpenSearch-Dashboards.git
-    ref: '2.4'
+    ref: '061e5bd8706c3a6a1aa089e6490c56397aa6b60a'
```
Comment (peterzhuamazon) [inline manifests/2.4.0/opensearch-dashboards-2.4.0.yml:12]: Mistake?
Could not find this ref at all.
https://github.com/opensearch-project/opensearch-build/actions/runs/3399364426/jobs/5654890776

Comment (zelinh) [reply manifests/2.4.0/opensearch-dashboards-2.4.0.yml:12]: Strange. Double checked the commit ID from https://ci.opensearch.org/ci/dbc/distribution-build-opensearch-dashboards/2.4.0/4275/linux/x64/tar/dist/opensearch-dashboards/manifest.yml and it's the same.

Comment (peterzhuamazon) [reply manifests/2.4.0/opensearch-dashboards-2.4.0.yml:12]: The code is trying to checkout out using --branch with --depth 1 which does not allow commitID it seems. Only a branch or tag is allowed.

Let me see if we can change the code a bit to allow a full clone then git checkout.

Comment (peterzhuamazon) [reply manifests/2.4.0/opensearch-dashboards-2.4.0.yml:12]: Interesting it works for the OS not the OSD one.
Need more debugging as this command is actually possible to checkout commits:
`git fetch --depth 1 origin 77ef9e304dd6ee95a600720a387a9735bbcf7bc9`

Comment (peterzhuamazon) [reply manifests/2.4.0/opensearch-dashboards-2.4.0.yml:12]: OS: https://github.com/opensearch-project/opensearch-build/actions/runs/3399364426/jobs/5654890705
OSD: https://github.com/opensearch-project/opensearch-build/actions/runs/3399364426/jobs/5654890776

Seems like it is additional checks here that cause the issue due to `git ls-remote` will only check for branch/tag.
` /home/runner/work/opensearch-build/opensearch-build/src/ci_workflow/ci_check_list_source_ref.py`

Comment (peterzhuamazon) [reply manifests/2.4.0/opensearch-dashboards-2.4.0.yml:12]: I fixed it in this PR @zelinh please review:
* https://github.com/opensearch-project/opensearch-build/pull/2856

**Comment by peterzhuamazon** [comment]
The ci checks fix PR needs to be merged before this one:
* https://github.com/opensearch-project/opensearch-build/pull/2856

File: manifests/2.4.0/opensearch-dashboards-2.4.0.yml (hunk 1 above)
Comment (zelinh) [reply manifests/2.4.0/opensearch-dashboards-2.4.0.yml:12]: Just have above PR merged and fetch the changes. Re-running the CIs now.

**Review by peterzhuamazon — APPROVED** [review]

**Comment by amsiglan** [comment]
Please use commit id - `026413a38c0197566cef052a8de6e8333489e82c` for security-analytics-dashboards-plugin. Thanks!
@peterzhuamazon @zelinh

**Comment by peterzhuamazon** [comment]
OS core change to 877f891f121a6889c736c424bdd8a205c0e8ad69
OS sec change to bca461296d1c54f49e4d139316c855f9ca37be26

**Review by bbarani — APPROVED** [review]

**Pull request merged**